client/changelist/changelist_test.go

```diff
@@ -29,3 +29,36 @@ func TestMemChangelist(t *testing.T) {
 	cs = cl.List()
 	assert.Equal(t, 0, len(cs), "List should be empty")
 }
+
+func TestMemChangeIterator(t *testing.T) {
+	cl := memChangelist{}
+	it, err := cl.NewIterator()
+	assert.Nil(t, err, "Non-nil error from NewIterator")
+	assert.False(t, it.HasNext(), "HasNext returns false for empty ChangeList")
+
+	c1 := NewTufChange(ActionCreate, "t1", "target1", "test/targ1", []byte{1})
+	cl.Add(c1)
+
+	c2 := NewTufChange(ActionUpdate, "t2", "target2", "test/targ2", []byte{2})
+	cl.Add(c2)
+
+	c3 := NewTufChange(ActionUpdate, "t3", "target3", "test/targ3", []byte{3})
+	cl.Add(c3)
+
+	cs := cl.List()
+	index := 0
+	it, _ = cl.NewIterator()
+	for it.HasNext() {
+		c, err := it.Next()
```

Comment:
From the relatively small number of custom iterator examples I've found, having a check for the next element seems to be more popular than not. It makes it clear whether the loop is ending because of an error state, or because all iterations were completed.